### ISSUE TITLE
TickProfiler officially broken on latest KCauldron

### ISSUE THREAD

**Issue by Zilacon** [MinimallyCorrect/TickProfiler#29]
When using the latest KCauldron build (Im using build 168) doing /profile shows this:

```
>profile e 7 5
[21:17:55 ERROR]: Failed to get field_147482_g in world Overworld/5
[21:17:55 ERROR]: Failed to get field_72996_f in world Overworld/5
[21:17:55 INFO]: 
Profiling for 7 seconds in Overworld/5
[21:18:02 INFO]: 
TPS: 19.965773
Single Entity | Time/Tick | %

Chunk | Time/Tick | %

All Entities of Type | Time/Tick | %

Average Entity of Type | Time/tick | Calls
```

A fix for this would be great.

**Comment by LunNova** [comment]
KCauldron changes the type of LoadedEntityList/LoadedTileEntityList to Queue<Entity>/Queue<TileEntity>.

I'd never heard of it before, and as far as I know it's not very common. I can't think of a good way of fixing it without having to duplicate quite a bit of code. Given that it doesn't seem very popular and I don't really want to spend the time on it, I won't be fixing this.

I will accept a pull request to fix this if someone makes one.

**Comment by Zilacon** [comment]
Every 1.7.10 modded server runs KCauldron nallar... Many modpacks wont even start up or run without KCauldron because of needing the most up to date FORGE which KCauldron is the only solution for.

Cauldron was discontinued, the author BloodMC is working on Sponge. KCauldron is the ONLY 1.7.10 solution for servers right now and TickProfiler is the ONLY server profiler right now.

**Comment by Zilacon** [comment]
Plus KCauldron is also included in Technic server packs as well. Like BlightFall, Tekkit Legends, etc.
ATL Is basically the same thing. Not sure about FTB i tend to stay away from their packs.

**Comment by LunNova** [comment]
Okay, I'll make a KCauldron specific 1.7.10 release since that's the
easiest way of fixing this.

**Comment by LunNova** [comment]
Experimental fix in 3db6fdc

Jenkins builds: http://nallar.me/buildservice/job/TickProfiler-1.7.10-KCauldron/

**Comment by Zilacon** [comment]
Thanks for agreeing to fix this.
I tried this version and i get these errors:
![yhzppfd](https://cloud.githubusercontent.com/assets/3002248/12080519/c8149e2a-b211-11e5-8e7f-f6230bb9c7d1.png)

**Comment by LunNova** [comment]
Could you try again with the newest build?

Also, just double checking, but definitely don't run this on a non-test server as there's a chance of bugs which screw up/wipe the TileEntity or Entity queues.

**Comment by Zilacon** [comment]
I got no way of testing this other than my live server. It would take me too long to transfer my live server to my desktop or re-create my modpack server on my desktop.

At the moment im trying to deal with my network, repair 4 customer computers, refurbish about 10 more to resell, posting ads on craigslist, and getting ready to post stuff on ebay as well. I simply dont got the time to make a test server lol.

**Comment by LunNova** [comment]
Can you suggest a small-ish modpack which includes a server setup using one of the newer versions of KCauldron, so I can test it?

**Comment by Zilacon** [comment]
Tekkit Legends is small. But technic is down so you cant download it, they are being DDoS'd bad for the last 4 days.

You could try Agrarian Skies 2? Its a small skyblock 1.7.10 modpack on CURSE.

**Comment by LunNova** [comment]
Seems to be working now, on Tekkit Legends. ![Screenshot](https://i.imgur.com/ZhOmnyY.png)

I recommend waiting until you can test it a bit (or at least having a recent world backup) in case it breaks everything, as I have only tested it for a few minutes on a 1 player world with barely anything in it.

**Comment by Zilacon** [comment]
Yeah ill test it out later on a live testing server, grab a few people to hop on.
And since i got you here could you add a world whitelist to nTweaks? I have tried manually unloading the massive list of worlds through console commands by adding multiworld world unload commands to the console scheduler plugin but that keeps crashing the server. However nTweaks did not crash the server when it was unloading the worlds.

But nTweaks unloads EVERY world even our spawn and overworlds. (We got 13 overworlds and a spawn world) if you could just add a world whitelist to the config so it ignores unloaading those worlds that would be great!

**Comment by LunNova** [comment]
I can add a whitelist, but I'm not sure why it's necessary. Worlds with any loaded chunks won't be unloaded. Do you ban/disable chunkloading items?

If there are chunkloaders but the worlds are unloading anyway then that's a bug.

**Comment by Zilacon** [comment]
I do not allow chunkloaders on my servers. However the spawn world is setup as the spawning point for the server and the world is loaded along the other 13 worlds by the KCauldron server and about 1 minute later nTweaks will unload every world but Dimension 0

**Comment by FiveEYZ** [comment]
how odd that it seems to be working perfectly fine for me using:

TickProfiler-1.7.10.40
along with 
KCauldron Version: pw.prok:KCauldron:1.7.10-1566.UNOFFICIAL

modpack: FTB Infinity Evolved 2.2.2

![3a5ad42f29d1a82166cad9dcd955cf25](https://cloud.githubusercontent.com/assets/9557442/12107569/194f5af8-b36d-11e5-9984-c34fff2a8b95.png)

**Comment by Zilacon** [comment]
This issue report was made in Oct 31st, you are using a build from December which was just recently released. There might have been something that changed in KC from oct to dec. Either way even nallar said it was not built to work with KCauldron so patching it to be compatible with it is still good either way.

**Comment by LunNova** [comment]
KCauldron _did_ need some different behaviour if you were running a build after [this commit.](https://gitlab.prok.pw/KCauldron/KCauldron/commit/9ba84893f398793dcaee8613bcb2f16238933835)

Those changes just got reverted in [this commit](https://gitlab.prok.pw/KCauldron/KCauldron/commit/f722fccdcea1a0faec61c1bd72d5dd56b0c17c14). 10/10: the day I get around to fixing it, the changes which broke it get undone.

**Comment by LunNova** [comment]
Closing this as:
- The fix in the special branch seemed to work
- It's no longer relevant as KCauldron just reverted the change which caused this issue. Any build after 1.7.10-1614.170 doesn't need the special branch.

**Comment by LunNova** [comment]
I will delete the kcauldron specific builds tomorrow to tidy up jenkins, unless someone objects.

The kcauldron specific code will still be available under tag 1.7.10-kcauldron (commit: 9aaebe383ee) in case someone needs it.

edit: And it's gone.
